.github/workflows/test.yml

```
@@ -41,25 +45,41 @@ jobs:
         # (copied from https://help.github.com/en/actions/migrating-to-github-actions/migrating-from-circleci-to-github-actions)
         run: |
           sudo chmod -R 777 $GITHUB_WORKSPACE /github /__w/_temp
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Install test dependencies
         run: |
           . /home/firedrake/firedrake/bin/activate
           python -m pip install pytest-timeout
           python -m pip install pytest-cov
+          # try and reduce the number of warnings to sift through
+          python -m pip install siphash24
```

Check the integration tests pass with the code-gen caching improvements.

```suggestion
          python -m pip install siphash24
          cd /home/firedrake/firedrake/src/fiat
          git fetch
          git checkout connorjward/fix-quadrature-hash
          cd -
          cd /home/firedrake/firedrake/src/firedrake
          git fetch
          git checkout connorjward/more-cache-fixes
          cd -
```